### PR TITLE
Forgot password

### PR DESCRIPTION
**Pull request by dmorina**
adds forgot password functionality
new button on login, sends email with reset_key or ignore link

### CODE REVIEW

**Comment by aditimithal** [comment]
I was trying to sign-in but it shows "Account is not activated "
On clicking the forgot password button it sends a link and and i am able to change the password too.But ,i am unable to login and it shows that the account is not activated. How do i go about it?
Also, will there be a message saying "invalid email address" if i try to add an invalid email-id ?

**Comment by dmorina** [comment]
have you received an activation link initially when you singed up?

**Comment by aditimithal** [comment]
I was an existing user so i did not register again :P So,
Entered Valid email-> I registered again , got the activation link and everything is working well :D
Invalid email-> I could not find a message that shows "invalid email". I hope the email i entered does not exist , because i chose a really weird name.(gmail account)

**Comment by aditimithal** [comment]
The email i chose does not exist(the test message bounced).

**Comment by dmorina** [comment]
ok, so the process is as follows you register, then you will get an activation link you follow it and it will activate your account, then you can use forgot password with the same email and you follow the other link :) it should work

**Comment by aditimithal** [comment]
That works if you have entered a valid email address :+1:

**Comment by nistala** [comment]
After I reques

**Comment by nistala** [comment]
clicked wrong button while commenting.

File: static/templates/authentication/forgot-password.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div  layout="row" flex="50" layout-align="center center">
+    <md-content class="authentication-form" layout-padding class="md-gray-theme" flex="50" layout-align="center center" layout="column">
+        <h3 id="form-header">Forgot Password</h3>
+        <form name="form">
+            <div layout="row">
+                <md-input-container>
+                    <label>Email</label>
+                    <input type="email" required ng-model="auth.email">
+                </md-input-container>
```
Comment (nistala) [inline static/templates/authentication/forgot-password.html]: after I submit the email address, i am on the same page, we should show something like "email sent" in a toast message or on the form.

**Comment by nistala** [comment]
Go to forgot password page
hit submit twice
click on first link and click enter new email and password
hit enter and I don't see an error message - ideally should say something like "looks like an expired request"

File: static/js/authentication/controllers/auth-settings.controller.js [hunk 2]
```diff
@@ -18,11 +18,18 @@
         var self = this;
 
         self.changePassword = changePassword;
+        self.resetPassword = resetPassword;
+        self.submitForgotPassword = submitForgotPassword;
 
         activate();
         function activate() {
             if (!Authentication.isAuthenticated() && $location.path().match(/change-password/gi)) {
```
Comment (nistala) [inline static/js/authentication/controllers/auth-settings.controller.js]: can we add a link for change-password on my account page

Comment (dmorina) [inline static/js/authentication/controllers/auth-settings.controller.js]: we will add it in the dropdown next to the username on top right corner

**Comment by maniksingh92** [comment]
lgtm functionality wise.

**Comment by dmorina** [comment]
@nistala fixed success message and the other issue with invalid reset_key

**Comment by nistala** [comment]
lgtm

**Comment by neilthemathguy** [comment]
lgtm

**Pull request merged**